### PR TITLE
[Feature](Kafka) Add kerberos support for kafka

### LINKED ISSUE #14434: [Bug] kafka kerberos authentication error 'missing a GSSAPI module' in routine load
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

master

### What's Wrong?

routine load fails for kafka cluster with kerberos authentication.

kafka kerberos authentication error 'missing a GSSAPI module' in routine load

### What You Expected?

routine load works as normal for kafka cluster with kerberos authentication.

### How to Reproduce?

1. prepare a kafka cluster with kerberos authentication.

2. create routine load with kerberos config

CREATE ROUTINE LOAD test_kerberos_routineload ON ttt3 
COLUMNS TERMINATED BY ",",  COLUMNS(id, name, age)  
PROPERTIES (
 "desired_concurrent_number"="3",
  "strict_mode" = "false"   )
FROM KAFKA (
     "kafka_broker_list" = "core-1-1.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092,core-1-2.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092,core-1-3.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092",
     "kafka_topic" = "test-kerboers-01",
     "property.securit
y.protocol" = "SASL_PLAINTEXT",
     "property.sasl.kerberos.service.name" = "kafka",
     "property.sasl.kerberos.keytab" = "/etc/krb5.keytab",
     "property.sasl.kerberos.principal" = "doris@EMR.C-6206E789B0D35587.COM" );

3. the following errors will raise

- `show routine load` will get `Failed to get all partitions of kafka topic:`

mysql> show routine load \G;
*************************** 1. row ***************************
                  Id: 11036
                Name: test_kerberos_routineload
          CreateTime: 2022-11-21 10:56:24
           PauseTime: 2022-11-21 10:56:37
             EndTime: NULL
              DbName: default_cluster:test
           TableName: ttt3
               State: PAUSED
      DataSourceType: KAFKA
      CurrentTaskNum: 0
       JobProperties: {"timezone":"Asia/Shanghai","send_batch_parallelism":"1","columnSeparator":"','","load_to_single_tablet":"false","lineDelimiter":"\n","maxBatchSizeBytes":"104857600","exec_mem_limit":"2147483648","strict_mode":"false","jsonpaths":"","currentTaskConcurrentNum":"0","fuzzy_parse":"false","partitions":"*","columnToColumnExpr":"id,name,age","maxBatchIntervalS":"10","whereExpr":"*","precedingFilter":"*","mergeType":"APPEND","format":"csv","json_root":"","deleteCondition":"*","desireTaskConcurrentNum":"3","maxErrorNum":"0","strip_outer_array":"false","execMemLimit":"2147483648","num_as_string":"false","maxBatchRows":"200000"}
DataSourceProperties: {"topic":"test-kerboers-01","currentKafkaPartitions":"","brokerList":"core-1-1.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092,core-1-2.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092,core-1-3.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092"}
    CustomProperties: {"security.protocol":"SASL_PLAINTEXT","kafka_default_offsets":"OFFSET_END","group.id":"test_kerberos_routineload_8ee13147-4507-4174-b211-0f129e878d2c","sasl.kerberos.keytab":"/etc/krb5.keytab","sasl.kerberos.principal":"doris@EMR.C-6206E789B0D35587.COM","sasl.kerberos.service.name":"kafka"}
           Statistic: {"receivedBytes":0,"runningTxns":[],"errorRows":0,"committedTaskNum":0,"loadedRows":0,"loadRowsRate":0,"abortedTaskNum":0,"errorRowsAfterResumed":0,"totalRows":0,"unselectedRows":0,"receivedBytesRate":0,"taskExecuteTimeMs":1}
            Progress: {}
                 Lag: {}
ReasonOfStateChanged: ErrorReason{code=errCode = 4, msg='Job failed to fetch all current partition with error errCode = 2, detailMessage = Failed to get all partitions of kafka topic: test-kerboers-01. error: Waited 5 seconds (plus 60593 nanoseconds delay) for io.grpc.stub.ClientCalls$GrpcFuture@4a351b23[status=PENDING, info=[GrpcFuture{clientCall={delegate=ClientCallImpl{method=MethodDescriptor{fullMethodName=doris.PBackendService/get_info, type=UNARY, idempotent=false, safe=false, sampledToLocalTracing=true, requestMarshaller=io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller@5dd46962, responseMarshaller=io.grpc.protobuf.lite.ProtoLiteUtils$MessageMarshaller@364601d, schemaDescriptor=org.apache.doris.proto.PBackendServiceGrpc$PBackendServiceMethodDescriptorSupplier@2f305194}}}}]]'}
        ErrorLogUrls:
            OtherMsg:
1 row in set (0.00 sec)

- `be.INFO` log error `Cyrus/libsasl2 is missing a GSSAPI module`

I1121 10:56:32.431066 20209 data_consumer.h:94] kafka log-2-LIBSASL, event: [thrd:sasl_plaintext://core-1-3.c-6206e789b0d35587.cn-beijing.emr.ali]: sasl_plaintext://core-1-3.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092/bootstrap: Cyrus/libsasl2 is missing a GSSAPI module: make sure the libsasl2-modules-gssapi-mit or cyrus-sasl-gssapi packages are installed
I1121 10:56:32.431089 20209 data_consumer.h:94] kafka log-3-FAIL, event: [thrd:sasl_plaintext://core-1-3.c-6206e789b0d35587.cn-beijing.emr.ali]: sasl_plaintext://core-1-3.c-6206e789b0d35587.cn-beijing.emr.aliyuncs.com:9092/bootstrap: Failed to initialize SASL authentication: SASL handshake failed (start (-4)): SASL(-4): no mechanism available: No worthy mechs found (after 0ms in state AUTH_REQ)

### Anything Else?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by amosbird**
# Proposed changes

Issue Number: close #14434  #14353

## Problem summary

Compile librdkafka with Kerberos SASL GSSAPI support. cc @xiaokang 

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [X] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [X] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [X] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [X] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [X] No

### CODE REVIEW

**Comment by xiaokang** [comment]
Thanks @amosbird for fixing the issuse!

A issue https://github.com/apache/doris/issues/14434 is created for others to search the problem. You can link to the issue.

File: thirdparty/build-thirdparty.sh [hunk 1]
```diff
@@ -817,9 +817,10 @@ build_cyrus_sasl() {
     check_if_source_exist "${CYRUS_SASL_SOURCE}"
     cd "${TP_SOURCE_DIR}/${CYRUS_SASL_SOURCE}"
 
-    CPPFLAGS="-I${TP_INCLUDE_DIR}" \
+    CFLAGS="-fPIC" \
+        CPPFLAGS="-I${TP_INCLUDE_DIR}" \
         LDFLAGS="-L${TP_LIB_DIR}" \
-        ./configure --prefix="${TP_INSTALL_DIR}" --enable-static --enable-shared=no --with-openssl="${TP_INSTALL_DIR}" --with-pic
+        ./configure --prefix="${TP_INSTALL_DIR}" --enable-static --enable-shared=no --with-openssl="${TP_INSTALL_DIR}" --with-pic --enable-gssapi="${TP_INSTALL_DIR}" --with-gss_impl=mit --with-dblib=none
```
Comment (xiaokang) [inline thirdparty/build-thirdparty.sh:823]: is '--with-dblib=none' needed?

File: thirdparty/build-thirdparty.sh [hunk 2]
```diff
@@ -836,6 +837,10 @@ build_librdkafka() {
 
     cd "${TP_SOURCE_DIR}/${LIBRDKAFKA_SOURCE}"
 
+    # NOTE(amos): librdkafka uses a weird autoconf variant (mklove) which doesn't allow extending PKG_CONFIG with spaces in cmd.
+    # As a result, we use a patch to hard code "--static" into PKG_CONFIG instead.
+    # PKG_CONFIG="pkg-config --static"
+
```
Comment (xiaokang) [inline thirdparty/build-thirdparty.sh:843]: do we need to add --enable-gssapi for build_librdkafka?

File: thirdparty/build-thirdparty.sh (hunk 1 above)
Comment (amosbird) [reply thirdparty/build-thirdparty.sh:823]: It helps exclude unneeded system dependencies (gdm in my case).

File: thirdparty/build-thirdparty.sh (hunk 2 above)
Comment (amosbird) [reply thirdparty/build-thirdparty.sh:843]: I don't think so. Linking to sasl2 will pull in all required archives.

**Comment by xiaokang** [comment]
LGTM

**Comment by xiaokang** [comment]
We have tested in an env with kafka kerberos authentication, the issue #14431 is fixed.

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**